sdk/servicebus/azservicebus/message.go

```diff
@@ -1,19 +1,108 @@
 package azservicebus
 
-import "github.com/Azure/azure-sdk-for-go/sdk/servicebus/azservicebus/internal"
+import (
+	"github.com/Azure/azure-amqp-common-go/v3/uuid"
+	"github.com/Azure/azure-sdk-for-go/sdk/servicebus/azservicebus/internal"
+	"github.com/Azure/go-amqp"
+)
+
+const (
+	annotationPartitionKey    = "x-opt-partition-key"
+	annotationViaPartitionKey = "x-opt-via-partition-key"
+)
 
 // Message represents a sendable message, including only user-settable properties.
 // A message that has been received from Service Bus will be of type `ReceivedMessage`.
 type Message struct {
-	Body []byte
-	ID   string
+	Body                    []byte
+	ID                      string
+	PartitionKey            *string
+	TransactionPartitionKey *string
+
+	// TODO: is an empty session ID valid?
+	SessionID string
 }
 
 // ReceivedMessage represents a message received from Service Bus.
 type ReceivedMessage struct {
 	Message
-	LockToken      string
-	SequenceNumber int64
+	LockToken      *string
```

Comment:
Is there a semantic difference between an empty string an `nil` here?  Similar question for `SequenceNumber` and `PartitionID` below.

Reply:
Let me check.

LockToken being an empty string is probably okay (I don't think it's a valid value in the range anyways).

Let me check on the other two. I'm thinking the zero values for them would be okay, in context, but want to see what we did with the other languages.

Reply:
Allright:
- LockToken is now just a string. We should be okay in any context where it matters.
- PartitionID was not actually supposed to be exported. So I've removed it.
- SequenceNumber is now just a plain int64, not a pointer.